pkg/vulnerability/vulnerability.go

```diff
@@ -28,31 +28,32 @@ func NewClient() Client {
 	}
 }
 
-func (c Client) FillAndFilter(vulns []types.DetectedVulnerability, severities []dbTypes.Severity,
-	ignoreUnfixed bool, ignoreFile string, light bool) []types.DetectedVulnerability {
+func (c Client) FillInfo(vulns []types.DetectedVulnerability, light bool) {
 	var err error
 	var severity dbTypes.Severity
 
-	ignoredIDs := getIgnoredIDs(ignoreFile)
-	var vulnerabilities []types.DetectedVulnerability
-	for _, vuln := range vulns {
-		var vulnerability dbTypes.Vulnerability
+	for i := range vulns {
 		if light {
-			severity, err = c.dbc.GetSeverity(vuln.VulnerabilityID)
-			vulnerability.Severity = severity.String()
+			severity, err = c.dbc.GetSeverity(vulns[i].VulnerabilityID)
+			vulns[i].Vulnerability.Severity = severity.String()
```

Comment:
What if `err` is not nil? Shall we log warn and skip it?

Reply:
This error is handled by line 43.
https://github.com/aquasecurity/trivy/pull/296/files#diff-d85a008c0dbe874b6d2b0caf6eee787cR43

```
		if light {
			severity, err = c.dbc.GetSeverity(vuln.VulnerabilityID)
			vulns[i].Vulnerability.Severity = severity.String()
		} else {
			vulnerability, err = c.dbc.GetVulnerability(vuln.VulnerabilityID)
		}
		if err != nil {
			log.Logger.Warn(err) // here
			continue
		}
```

I commonalized each error handling because they are the same, but do you prefer splitting like the following? You can decide it because I don't have a strong policy here.

```
		if light {
			severity, err = c.dbc.GetSeverity(vuln.VulnerabilityID)
			if err != nil{
				log.Lgger.Warn(err)
				continue
			}
			vulns[i].Vulnerability.Severity = severity.String()
		} else {
			vulnerability, err = c.dbc.GetVulnerability(vuln.VulnerabilityID)
			if err != nil{
				log.Lgger.Warn(err)
				continue
			}
		}
```

Reply:
Ah got it. Then it's fine. Let's maybe just add some context to the warning message, e.g. `log.Logger.Warn("Error while getting vulnerability details")`

Reply:
added